File: android/app/src/main/java/com/example/pametneucionice/network/RoomApiService.kt
package com.example.pametneucionice.network

import com.example.pametneucionice.model.RoomResponse
import retrofit2.http.GET
import retrofit2.http.Path

interface RoomApiService {

    @GET("api/rooms")
    suspend fun getRooms(): List<RoomResponse>

    @GET("api/rooms/{roomId}")
    suspend fun getRoom(@Path("roomId") roomId: String): RoomResponse
}

File: angular/src/app/services/room-api.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { RoomResponse, SensorUpdateRequest } from '../models/room.models';

@Injectable({ providedIn: 'root' })
export class RoomApiService {
  private readonly httpClient = inject(HttpClient);
  private readonly baseUrl = 'http://localhost:8080';

  getRooms(): Observable<RoomResponse[]> {
    return this.httpClient.get<RoomResponse[]>(`${this.baseUrl}/api/rooms`);
  }

  getRoom(roomId: string): Observable<RoomResponse> {
    return this.httpClient.get<RoomResponse>(`${this.baseUrl}/api/rooms/${roomId}`);
  }

  updateSensors(roomId: string, request: SensorUpdateRequest): Observable<RoomResponse> {
    return this.httpClient.put<RoomResponse>(`${this.baseUrl}/api/rooms/${roomId}/sensors`, request);
  }
}
